### PR TITLE
chore(#142): adding tests to increase coverage

### LINKED ISSUE #142: Improve unit test coverage for OpenMRS Mediator
Improve unit test coverage for openmrs

```
-----------------------------|---------|----------|---------|---------|---------------------------------------------------------------------------
File                         | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                                         
-----------------------------|---------|----------|---------|---------|---------------------------------------------------------------------------
All files                    |   71.78 |    66.25 |    62.5 |   71.78 |
 controllers                 |   67.46 |    66.66 |      75 |   67.46 |
  cht.ts                     |      50 |    33.33 |   66.66 |      50 | 21-22,28-51,58-59,61-63,73-82
  service-request.ts         |     100 |      100 |     100 |     100 |
 controllers/tests           |     100 |      100 |     100 |     100 |
  utils.ts                   |     100 |      100 |     100 |     100 |
 mappers                     |   62.78 |    27.77 |   55.55 |   62.78 |
  cht.ts                     |   62.35 |       25 |      60 |   62.35 | 106-146,148-170
  openmrs.ts                 |   63.54 |    33.33 |      50 |   63.54 | 40-64,70-78,87
 middlewares                 |     100 |      100 |     100 |     100 |
  index.ts                   |     100 |      100 |     100 |     100 |
 middlewares/schemas         |     100 |      100 |     100 |     100 |
  encounter.ts               |     100 |      100 |     100 |     100 |
  endpoint.ts                |     100 |      100 |     100 |     100 |
  organization.ts            |     100 |      100 |     100 |     100 |
  patient.ts                 |     100 |      100 |     100 |     100 |
  service-request.ts         |     100 |      100 |     100 |     100 |
 middlewares/schemas/tests   |     100 |      100 |     100 |     100 |
  cht-request-factories.ts   |     100 |      100 |     100 |     100 |
  fhir-resource-factories.ts |     100 |      100 |     100 |     100 |
 routes                      |     100 |      100 |     100 |     100 |
  cht.ts                     |     100 |      100 |     100 |     100 |
  encounter.ts               |     100 |      100 |     100 |     100 |
  endpoint.ts                |     100 |      100 |     100 |     100 |
  organization.ts            |     100 |      100 |     100 |     100 |
  patient.ts                 |     100 |      100 |     100 |     100 |
  service-request.ts         |     100 |      100 |     100 |     100 |
 utils                       |    62.1 |    64.42 |   61.22 |    62.1 |
  cht.ts                     |   53.08 |    28.57 |   54.54 |   53.08 | 38-40,59-85,88-103,116-119,127-129,132-141,143-152,160-162
  fhir.ts                    |   71.61 |    71.11 |   77.77 |   71.61 | 112-114,140-147,177-179,182-210,212-223,225-236
  openhim.ts                 |     100 |      100 |     100 |     100 |
  openmrs.ts                 |   26.38 |      100 |       0 |   26.38 | 17-22,24-52,54-62,64-72
  openmrs_sync.ts            |   61.65 |    66.66 |   53.84 |   61.65 | 37-38,44-45,95-96,107-108,177-181,187-196,204-235,241-245,254-301,310-326
  request.ts                 |     100 |      100 |     100 |     100 |
  url.ts                     |     100 |      100 |     100 |     100 |
-----------------------------|---------|----------|---------|---------|---------------------------------------------------------------------------
```

src/controllers/cht.ts

- [ ] updatePatientIds
- [x] createPatient where source id is specified
- [ ] patient is not found
- [ ] encounter w/ observations

src/mappers/cht.ts

- [ ] observations (covered by encounter w/ observations)

src/mappers/openmrs.ts

- [ ] visitNote
- [ ] observations

src/utils/cht.ts

- [ ] getPatientUUIDFromSourceId (covered by updatePatientsIds)
- [ ] getLocation
- [ ] createChtPatient (covered elsewhere, add specific test)
- [ ] chtRecordFromObservations
- [ ] queryCht
- [ ] chtRecordsApi (error branch) 

src/utils/fhir.ts

- [ ] addId
- [ ] error branches (createFhirPatient, createFhirResource)
- [ ] getFhirResourceSince

src/utils/openmrs.ts
- [ ] getOpenMRSPatientResource
- [ ] getOpenMRSResourcesSince
- [ ] createOpenMRSResource
- [ ] updateOpenMRSResource

src/utils/openmrs_sync.ts

- [ ] getResources (error branch)
- [ ] compare (invalid date)
- [ ] compare (error branch)
- [ ] getPatient
- [ ] getObservations
- [ ] sendEncounterToOpenMRS
- [ ] sendObservationToFhir
- [ ] sendEncounterToFhir
- [ ] syncEncounters

### PR DESCRIPTION
**Pull request by witash**
closes #142

### CODE REVIEW

File: mediator/src/utils/tests/fhir.spec.ts
Comment (lorerod) [inline mediator/src/utils/tests/fhir.spec.ts:1]: The only function in fhir.ts that has not been tested here is getFhirResource, which is never used. So, no tests are needed.

File: mediator/src/utils/tests/openmrs_sync.spec.ts [hunk 1]
```diff
@@ -1,149 +1,261 @@
...
-    });
-    jest.spyOn(openmrs, 'getOpenMRSResourcesSince').mockResolvedValueOnce({
-      data: [
-        { id: 'incoming', ...constants },
-        { id: 'toupdate', ...constants }
-      ],
-      status: 200,
-    });
+  describe('compare', () => {
+    it('compares resources with the gvien key', async () => {
```
Comment (lorerod) [inline mediator/src/utils/tests/openmrs_sync.spec.ts:23]:
```suggestion
    it('correctly identifies incoming, outgoing, and to-be-updated resources based on the given key', async () => {
```

**Review by lorerod — APPROVED** [review]
Thanks, @witash! This looks great to me! I’ve added a couple of suggestions for improvement, but there are no blockers.

File: mediator/src/utils/tests/cht.spec.ts [hunk 2]
```diff
@@ -36,4 +44,86 @@ describe('CHT Utils', () => {
...
   });
+
+  describe('getLocationFromOpenMRSPatient', () => {
+    it('should return place ID if address contains place ID', async () => {
+      const fhirPatient = OpenMRSPatientFactory.build({}, {
+        addressKey: 'address4',
+        addressValue: 'FCHV Area [12345]'
+      });
+
+      const result = await getLocationFromOpenMRSPatient(fhirPatient as any);
```
[the remark this answers is not in the document]
Comment (witash) [reply mediator/src/utils/tests/cht.spec.ts:55]: no it's not, good catch

File: mediator/src/utils/tests/cht.spec.ts
[the remark this answers is not in the document]
Comment (witash) [reply mediator/src/utils/tests/cht.spec.ts:1]: added tests for getPatientUUIDFromSourceId

**Comment by witash** [comment]
@lorerod didn't see your comments before merging, but the changes are here https://github.com/medic/cht-interoperability/commit/e3f1a4c3c65578f1cc8ee778be2eeecad78743ad

**Pull request merged**